### ISSUE TITLE
WEBMs randomly stop loading/playing

### ISSUE THREAD

**Issue by reportthings** [K1rakishou/Kuroba-Experimental#699]
Randomly, for an indeterminate amount of time, WEBMs will not load/play. Using MPV.  Restarting app and phone does not help.  The WEBMs load instantly in browser. Sometimes it lasts for 5 minutes, sometimes 5 hours.  All gifs and images load as normal.  

If you need any information or logs I am happy to provide.
And as always thanks for all the work you do on this app.

Here is a video showing what's happening:

https://user-images.githubusercontent.com/85461551/147840789-926a3703-8f2f-40c3-a10f-ef272cc019db.mp4

**Comment by reportthings** [comment]
Some more info: I said above that restarting app and phone did not help, but after trying both of those, I decided to try "Force Stopping" the app. 

After force stopping the app, WEBMs began to play as normal.  Though I can't be certain that actually resolved it, or if it resolved itself as it normally does after a random period of time.

**Comment by K1rakishou** [comment]
Yeah I'm gonna need the logs.
The best way would be to grab them via adb so that nothing is lost. 
Here is some info on how to install adb and stuff https://github.com/K1rakishou/Kuroba-Experimental/wiki/Getting-application-logs-or-crashlogs-on-beta-or-release-builds

**Comment by reportthings** [comment]
Just updating to let you know I do plan on uploading logs. I just do not have my computer with me while visiting family for the new year.  But I will update when I am back at home.  I do not wish to put any burden on you. 

By the end of January 2, or 3 I will be able to get the logs to you.

**Comment by K1rakishou** [comment]
It's fine. Just post them when you have time.

**Comment by reportthings** [comment]
Here are the logs, it's the best I can do I think.  For some reason ADB does not play well with android 11 but I managed to grab a real time log of the issue occurring with logcat.  I opened a couple hundred webms in a row, mostly swiping through catalog mode, opening some manually.   The issue began I think between line 1100 and line 1200.  I still included the earlier lines in case there is some kind of problem that occurs earlier than when the problem actually arose.  

 While watching the CMD show all my actions, there didn't seem to appear to be anything different about the lines where WEBMs stopped loading, and lines where WEBMs loaded fine, but it's possible I could have missed that.


[KurobaExLog.txt](https://github.com/K1rakishou/Kuroba-Experimental/files/7804660/KurobaExLog.txt)
 
Sorry if these weren't the logs you were exactly after but neither version of 
https://github.com/K1rakishou/Kuroba-Experimental/wiki/Getting-application-logs-or-crashlogs-on-beta-or-release-builds instructions would work with android 11 unfortunately.

**Comment by K1rakishou** [comment]
Yeah, for some reason there is only like 1% of the app logs there for some reason and 0% logs from mpv. Just tried the same (adb logcat) on the emulator and it works fine.

**Comment by K1rakishou** [comment]
I guess the only thing I can try doing is updating the mpv to the latest version (the current build is a couple of months old). Also do you experience the same problems when using exoplayer?

**Comment by reportthings** [comment]
Sorry for giving you bad logs, I will install an emulator this evening and replicate the issue and try to provide proper logs.

As far as exoplayer, there are some possibly notable discoveries I noticed after trying to recreate the issue with it:

Exoplayer does not exhibit the issue
After trying to recreate the issue with exoplayer, I recreated it with MPV again.  I found that if the issue occurs with MPV, and I switch to exoplayer, the error "decoder init failed" appears on every WEBM and they fail to play with exoplayer as well.  

But only if the issue is first produced by MPV.  I tried almost 500 WEBMs in exoplayer, and besides the occasional "decoder init failed" that has always been present, no further issues existed.

The full "decoder init failed" message is:

`Failed to load video (reason: Decoder init failed: c2.android.vp8.decoder, Format(1, null, null, video/x-vnd.on2.vp8, null, -1, und, [1280, 720, -1.0], [-1, -1]))`

**Comment by K1rakishou** [comment]
>decoder init failed

>Failed to load video (reason: Decoder init failed: c2.android.vp8.decoder, Format(1, null, null, video/x-vnd.on2.vp8, null, -1, und, [1280, 720, -1.0], [-1, -1]))

Seems like that might be the reason why videos stop playing. But in that case I don't think I will be able to do anything.

**Comment by reportthings** [comment]
Ah, so you think that the 'decoder init failed' is the issue for MPV but is just not visible when using MPV? I guess that would make sense but I'm a little out of my depth when it comes to this stuff.  Feel free to close the issue if that's the case.  But if you still have doubts I'll try to produce proper logs with adb tonight and post them here.

**Comment by K1rakishou** [comment]
I don't catch and display mpv errors like I do with ExoPlayer errors. I probably should do that.
I still want to see the logs, this may be a bug in mpv so logs might help fix it.
I'm also planning to update the mpv and all the dependencies to the latest version in the next app update so it might be a good idea to first wait for that and see if it changes anything.

**Comment by K1rakishou** [comment]
Oh by the way. mpv supports software decoders. You can try switching from HW decoders to SW decoders and see whether it helps or not.

**Comment by reportthings** [comment]
Thanks. I'll try both HW and SW when I get a chance and update if I'm able to get useful logs.  If not I'll wait for the update and test again.  Thanks for your time

**Comment by reportthings** [comment]
Ok so after you pushed the beta build with the updated MPV and I reinstalled dependencies I have tested a couple dozen times.   The results are:

I can no longer replicate the bug I was experiencing, so all seems fine now.  I've tried about 20 times using my old method of swiping through WEBM catalogs on /gif/ to quickly load and close mass amounts of WEBMs and haven't had a problem.  Going to close this issue now.  It's fixed.

Thanks again for taking the time to help with my issues and for being so quick to update when there are problems.  Take care
